CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs

```diff
@@ -0,0 +1,301 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CrossViewEditorRibbonPart.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru.
+//
+//    This file is part of CDP4-IME Community Edition.
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4CrossViewEditor
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reactive.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading.Tasks;
+    using CDP4Common.EngineeringModelData;
+    using CDP4Composition;
+    using CDP4Composition.Navigation;
+    using CDP4Composition.Navigation.Interfaces;
+    using CDP4Composition.PluginSettingService;
+    using CDP4Dal;
+    using CDP4Dal.Events;
+    using CDP4OfficeInfrastructure;
+    using NLog;
+    using ReactiveUI;
+
+    /// <summary>
+    /// The purpose of the <see cref="CrossViewEditorRibbonPart"/> class is to describe and provide a part of the Fluent Ribbon
+    /// that is used in an Office addin. A <see cref="RibbonPart"/> always describes a ribbon group containing different controls
+    /// </summary>
+    public class CrossViewEditorRibbonPart : RibbonPart
+    {
+        /// <summary>
+        /// The NLog logger
+        /// </summary>
+        private static Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// The <see cref="IOfficeApplicationWrapper"/> that provides access to the loaded Office application
+        /// </summary>
+        private readonly IOfficeApplicationWrapper officeApplicationWrapper;
```

Comment:
Code smell:
> Remove this unread private field 'officeApplicationWrapper' or refactor the code to use its value.

Is this needed?

Reply:
leave it for now. It might become relevant when business logic is there.

Reply:
Yes. It's needed in the near future.